src/Olm.cpp

```diff
@@ -71,6 +77,43 @@ handle_to_device_messages(const std::vector<mtx::events::collections::DeviceEven
                                   e.what(),
                                   j_msg.dump(2));
                         }
+                } else if (msg_type == to_string(mtx::events::EventType::KeyVerificationAccept)) {
```

Comment:
Verification messages could be encrypted, can you change this to

a) not convert the message to json just to get the type, then convert every type to a string and compare them as strings? It is probably easier to just get the type via a visit or visit each type of message via overloaded or similar.
b) Handle encrypted KeyVerificationMessages, i.e. decrypt a message, if it is encrypted and then handle the decrypted message?

Thanks :3